pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -621,34 +638,44 @@ pub mod pallet {
 		type Message = GatewayMessage<T::Message, T::RouterId>;
 
 		fn process(msg: Self::Message) -> (DispatchResult, Weight) {
-			match msg {
-				GatewayMessage::Inbound {
-					domain_address,
-					message,
-					router_id,
-				} => {
-					let mut counter = 0;
-
-					let res = Self::process_inbound_message(
+			// The #[transactional] macro only works for functions that return a
+			// `DispatchResult` therefore, we need to manually add this here.
+			with_transaction(|| {
```

Comment:
We do not need this if we choose to remove the inbound batch support, right?

Reply:
We do, we still need to rollback storage in case of a failure.

Reply:
That's right!